src/client/dfs/dfs.c

```diff
@@ -63,9 +63,9 @@
 #define MODE_IDX	0
 #define OID_IDX		(sizeof(mode_t))
 #define ATIME_IDX	(OID_IDX + sizeof(daos_obj_id_t))
-#define MTIME_IDX	(ATIME_IDX + sizeof(time_t))
-#define CTIME_IDX	(MTIME_IDX + sizeof(time_t))
-#define CSIZE_IDX	(CTIME_IDX + sizeof(time_t))
+#define MTIME_IDX	(ATIME_IDX + sizeof(uint64_t))
+#define CTIME_IDX	(MTIME_IDX + sizeof(uint64_t))
+#define CSIZE_IDX	(CTIME_IDX + sizeof(uint64_t))
```

Comment:
This implies this works as expected, right?
```
time_t mtime = ...;
uint64_t mtime_hlc = (uint64_t) mtime;
```
Meaning, if we read an older container that stored times as `time_t`, but into a `uint64_t`, it's the same time?

Reply:
The suggestion is we squash the two PRs at/before landing so there is no "older" container.

Reply:
What about a 2.0 container?

Reply:
Reviewing this now, the key is in entry_stat in https://github.com/daos-stack/daos/pull/9636/files

There is code there to read the values the old way if there is no mtime handling so I think it's sufficient that these types are the correct size.

Reply:
I have double-checked now and yes they are.